sklearn/metrics/_pairwise_distances_reduction/__init__.py

```diff
@@ -45,46 +46,49 @@
-#          (dispatcher)                                   (dispatcher)
-#            ArgKmin                                     RadiusNeighbors
-#               |                                              |
-#               |                                              |
-#               |              (float{32,64} implem.)          |
-#               |           BaseDistancesReduction{32,64}      |
-#               |                       ∆                      |
-#               |                       |                      |
-#               |                       |                      |
-#               |     +-----------------+-----------------+    |
-#               |     |                                   |    |
-#               |     |                                   |    |
-#               x     |                                   |    x
-#            ArgKmin{32,64}                        RadiusNeighbors{32,64}
-#               |     ∆                                   ∆    |
-#               |     |                                   |    |
-#        ======================= Specializations =============================
-#               |     |                                   |    |
-#               |     |                                   |    |
-#               x     |                                   |    x
-#        EuclideanArgKmin{32,64}               EuclideanRadiusNeighbors{32,64}
+#                                      (base dispatcher)
+#                               BaseDistancesReductionDispatcher
+#                                              ∆
+#                                              |
+#                                              |
+#                              +---------------+---------------+
+#                              |                               |
+#                         (dispatcher)                    (dispatcher)
+#                           ArgKmin                      RadiusNeighbors
+#                              |                               |
+#                              |                               |
+#                              |     (float{32,64} implem.)    |
+#                              | BaseDistancesReduction{32,64} |
+#                              |               ∆               |
+#                              |               |               |
+#                              |               |               |
+#                              |    +----------+----------+    |
+#                              |    |                     |    |
+#                              |    |                     |    |
+#                              x    |                     |    x
+#             +-----------⊳ ArgKmin{32,64}         RadiusNeighbors{32,64} ⊲-------+
+#             |                |    ∆                     ∆    |                   |
+#   ArgKminClassmode{32,64}    |    |                     |    |   RadiusNeighborsClassMode{32,64}  # noqa: E501
```

Comment:
`ArgKminClassMode` is the dispatcher for `ArgKminClassmode{32, 64}`. I'm not sure where to put that information in the diagram, without getting too confusing.

Reply:
I proposed something in a19810332a4e59100e39ca1b0253f00e92d4d5b3.